packages/react/src/components/form/form.stories.tsx

```diff
@@ -0,0 +1,134 @@
+
+export default {
+  title: 'React/Form',
+  component: Form,
+} as Meta<FormProps<unknown>>;
+
+interface Values {
+  firstName: string;
+  lastName: string;
+  email: string;
+  day?: number;
+  month?: number;
+  year?: number;
+  source?: string;
+  agreement: boolean;
+}
+
+export const Playground: Story<FormProps<Values>> = (props) => (
+  <Form {...props}>
+    <Field>
+      <Input name="firstName" required>
+        First name <Asterisk aria-label="required" />
+      </Input>
+      <Validation state="error" for="valid">
+        Please fill in this field
+      </Validation>
+    </Field>
+
+    <Field>
+      <Input name="lastName" required>
+        Last name <Asterisk aria-label="required" />
+      </Input>
+      <Validation state="error" for="valid">
+        Please fill in this field
+      </Validation>
+    </Field>
+
+    <Field>
+      <Input type="email" name="email" required>
+        Email address <Asterisk aria-label="required" />
+      </Input>
+      <Validation state="error" for="valueMissing">
+        Please fill in this field
+      </Validation>
+      <Validation state="error" for="typeMismatch">
+        Please enter a valid email address
+      </Validation>
+    </Field>
+
+    <Field>
+      <FieldLabel htmlFor="dob-day">
+        Date of birth
+        <HelperText>DD / MM / YYYY</HelperText>
+      </FieldLabel>
+      <div
+        style={{
+          display: 'grid',
+          gap: '0.5rem',
+          grid: '1fr / 5rem 5rem 6rem',
+        }}
+      >
+        <Input
+          id="dob-day"
+          type="number"
+          name="day"
+          placeholder="DD"
+          min={1}
+          max={31}
+        />
+        <Input type="number" name="month" placeholder="MM" min={1} max={12} />
+        <Input
+          type="number"
+          name="year"
+          placeholder="YYYY"
+          min={new Date().getUTCFullYear() - 150}
+          max={new Date().getUTCFullYear()}
+        />
+      </div>
+    </Field>
+
+    <Field>
+      <FieldLabel>How did you hear about us?</FieldLabel>
+      <ul
+        style={{
+          listStyle: 'none',
+          margin: 0,
+          padding: 0,
+        }}
+      >
+        <li>
+          <Radio name="source">
+            Social media
+            <HelperText>Facebook, Twitter, TikTok, etc...</HelperText>
+          </Radio>
+        </li>
+        <li>
+          <Radio name="source">Search engine</Radio>
+        </li>
+        <li>
+          <Radio name="source">Word of mouth</Radio>
+        </li>
+        <li>
+          <Radio name="source">Other...</Radio>
+        </li>
+      </ul>
+    </Field>
+
+    <Field>
+      <Checkbox name="agreement" required>
+        I agree to the <a>Privacy policy</a> and <a>Terms and conditions</a>.
+      </Checkbox>
+      <Validation state="error" for="valid">
+        Please confirm you agree
+      </Validation>
+    </Field>
+
+    <Button>Send email</Button>
```

Comment:
One thing to note, that now disabling the form does not disable the button for submission.

Would you expect such behaviour?

Reply:
Yes, disabled buttons are an anti-pattern, you should be able to try to submit the form, but get validation messages that prevent you. Just like pressing enter in any field.

We can also ask @Subcide for his opinion here, but I remember him being against disabled buttons.

Reply:
Here's one use case, a Stepper pattern where each Step is a Form.
Steps other than the current gets disabled.

![image](https://user-images.githubusercontent.com/18623773/110653016-cee2d280-81b4-11eb-8e4f-3d5acdb03de0.png)

You still want buttons to interact with the Form though.

Reply:
You do have a point though, because `disabled` fields in a `Form` are simply ignored.

Which is why there's a `if (disabled) return;` in `Form`'s `onSubmit` which essentially prevents disabled `Form`s from being submitted even if the button isn't disabled.

Still, clicking the button should give you some form of feedback, rather than disabling the interaction altogether.

This is now stepping more into design territory though.

Reply:
Yeah probably disabling a form should then be combined with (future) "loading" button state, that also is disabling clicking that button...

Design if your form though to me looks like a bit odd UX... I don't think "disabled" state of form elements is meant to be used like such, but what do I know... 😓

Reply:
Oh I agree, it is weird. The step vanishing from view while you work on the next would be more usual.

Even with a loading state button, I still think we should not disable it and continue to emit events.

It's up to the consumer then to decide what to do with the `submit` events, in case it's loading.

They already control whether the button is loading or the form disabled after all.

Reply:
So after some investigation I would agree that `button` inside `form` should be disabled, as should `input` and `select`.
That's what `fieldset` does after all.
Additionally `<a>` inside `fieldset` have `pointer-events: none;` set, so we should probably do the same for `form`.

If the customer wants an enabled button inside a disabled form they can either:
- manually set it `<Button disabled={false} />` 
- put the button outside of the form
- use a `fieldset` to disable other fields, but not button

Are you fine with the idea that `<Form disabled />` has the same behavior as `<fieldset disabled />`?
If so, I'll make it so `Button` also uses the `form` hook.